File: templates/tests-php/BREACH-004.php
<?php

namespace Compliance\Tests;

use PHPUnit\Framework\TestCase;

class DataResidencyTest extends TestCase
{
    public function testRejectForeignDataResidency(): void
    {
        $config = Config::load();
        $residency = $config->infrastructure->dataResidency;
        $this->assertContains($residency, ['Local', 'Hybrid']);
    }
}

File: templates/tests-csharp/BREACH-004.cs
using Xunit;

namespace Compliance.Tests
{
    public class DataResidencyTests
    {
        [Fact]
        public void TestRejectForeignDataResidency()
        {
            var config = Config.Load();
            Assert.Contains(config.Infrastructure.DataResidency, new[] { "Local", "Hybrid" });
        }
    }
}
